lib/skate_web/router.ex

```diff
@@ -35,4 +53,21 @@ defmodule SkateWeb.Router do
     get "/routes", RouteController, :index
     get "/routes/:route_id", RouteController, :show
   end
+
+  defp put_user_token(conn, _) do
+    token = Guardian.Plug.current_token(conn)
+    assign(conn, :user_token, token)
+  end
+
+  defp api_auth(conn, _) do
+    with [token] <- get_req_header(conn, "token"),
+         {:ok, _claims} <- AuthManager.decode_and_verify(token) do
```

Comment:
What are your API requests, and are they related to Cognito somehow?

In signs_ui our API authentication (the requests that update the countdown signs) doesn't use the Cognito pool, since there's no underlying user to redirect to the auth page and back.

While I think it's possible to use Guardian for both cases, right now the `ErrorHandler` seems to be set up only to work with Cognito. But for us, for our "API" requests, since we control the software on both ends, we just compare the value here to an ENV var.

Reply:
Our API requests are for all the underlying data, so we only want authorized users to be able to access the results. I'm I'm understanding the question correctly they are related to Cognito in that we want to use the JWT token from Cognito to verify that this user is logged in (and thus an MBTA employee). We do take the user to an Active Directory login page before they can access the page to make the API requests. Does that answer the question?

Reply:
Oh, I think I misunderstood the flow. 

So are these API requests because of the SPA architecture, and are conceptually part of the session of the logged in user? Like, they visit the URL, and if they're authenticated it will load the page, which will start up the JS app, which then makes these requests because of actions the user takes?

If that's the case, does it make sense to just use `:ensure_auth` here, since it will be sending the cookie anyway? The alternative you're using here, of using a Bearer token is useful when supporting, say, mobile apps, or sites on multiple domains or the like.

Reply:
@losvedir your `:browser` suggestion help me find a solution to this, thanks! I just needed to split out the `accepts` plugs.